pipeline/container/manager.py

```diff
@@ -61,25 +57,38 @@ def __init__(self, pipeline_path: str):
                     f" {self.pipeline_module_str}"
                 )
             )
-        except Exception as e:
-            raise ValueError(f"Unexpected error: {e}")
 
         self.pipeline_name = os.environ.get("PIPELINE_NAME", "unknown")
         self.pipeline_image = os.environ.get("PIPELINE_IMAGE", "unknown")
 
         logger.info(f"Pipeline set to {self.pipeline_path}")
 
+    def __init__(self, pipeline_path: str):
+        self.pipeline_state: pipeline_schemas.PipelineState = (
+            pipeline_schemas.PipelineState.not_loaded
+        )
+        self.pipeline_state_message: str | None = None
+
+        try:
+            self._load(pipeline_path)
+        except Exception:
+            self.pipeline_state = pipeline_schemas.PipelineState.load_failed
+            tb = traceback.format_exc()
+            self.pipeline_state_message = tb
+            return
+
     def startup(self):
+        if self.pipeline_state == pipeline_schemas.PipelineState.load_failed:
+            return
         # add context to enable fetching of startup logs
         with logger.contextualize(pipeline_stage="startup"):
```

Comment:
maybe we could add similar to the `_load()` function so that we can show the full logs to the user?

Reply:
Ye that sounds even better tbh.

Reply:
So we'd have to update this line in pcore to query for `startup|loading` if we want those logs to come through, so perhaps worth adding a ticket for that?

Reply:
done https://linear.app/mystic-ai/issue/PC-998/allow-loading-logs